### DISCUSSION TITLE
Issue with default collation(?) when inserting data

### DISCUSSION THREAD

**Discussion by PhilBookst** [planetscale/discussion#114] [Bug Reports] [answered]
Hey, 

I'm trying to insert these 2 lines:

INSERT INTO hashtag (id, name, thumbnail, post_count) VALUES (NULL, 'neslitagül', NULL, NULL);
INSERT INTO hashtag (id, name, thumbnail, post_count) VALUES (NULL, 'neslitagul', NULL, NULL);

But I get the following error: 
rpc error: code = AlreadyExists desc = Duplicate entry 'neslitagul' for key 'hashtag.PRIMARY' (errno 1062) (sqlstate 23000) (CallerID: unsecure_grpc_client): Sql: "insert into hashtag(id, `name`, thumbnail, post_count) values (null, :vtg1, null, null)", BindVars: {vtg1: "type:VARBINARY value:\"neslitagul\""}

I then checked my db character_set (utf8mb4)  and collation_database (utf8mb4_0900_ai_ci)
and I guess the error stems from the ai in utf8mb4_0900_ai_ci, since it causes accent insensitivity?

Or rather because of the VARBINARY type?

Is there any way to fix this?

Thank you!

**Comment by samlambert** [comment]
Can you share the schema of this table?

**Reply by PhilBookst** [reply]
```
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["referentialIntegrity"]
}

datasource db {
  provider             = "mysql"
  url                  = env("DATABASE_URL")
  shadowDatabaseUrl    = env("SHADOW_DATABASE_URL")
  referentialIntegrity = "prisma"
}

model Hashtag {
  id            String?         @unique
  name          String          @id
  thumbnail     String?
  post_count    Int?
  images        Image[]

  @@map("hashtag")
}

model Image {
  id              String           @id
  thumbnail       String
  shortcode       String           @unique
  like_count      Int
  comment_count   Int
  caption         String?          @db.Text // for long captions
  hashtag         Hashtag?         @relation(fields: [hashtagName], references: [name])
  hashtagName     String?
  
  @@index([hashtagName])
  @@map("image")
}
```

I'm using prisma for my schema.

I'm trying to transfer data from a postgres db to planetscale and got the pgdump with --data-only and --column-inserts working by simply changing true/false to 1/0 respectively and then piping the file into pscale shell if that matters.

**Comment by jgreet** [comment, marked as answer]
To convert that existing table to [accent sensitive collation](https://dev.mysql.com/doc/refman/8.0/en/charset-collation-names.html) you could create a [new database branch](https://docs.planetscale.com/concepts/branching), run the following command in `pscale shell`, and then deploy the schema change to your main branch.

```sql
alter table hashtag convert to character set utf8mb4 collate utf8mb4_0900_as_ci;
```

I'm not certain if Prisma offers a way to change collations through migrations or not.

**Reply by PhilBookst** [reply]
Awesome now I can insert the 2 lines, thank you!

Is there a way to set this for tables in my db or just do it by hand every time I create a new table?

**Reply by jgreet** [reply]
You'll have to handle setting the collation for every table that needs it. We don't currently support the `ALTER DATABASE` command you'd need to change it at the database level.

We should take a look at allowing that sort of change though, maybe it's something we can expose in the future.
